graphql_compiler/compiler/ir_lowering_match.py

```diff
@@ -578,6 +593,194 @@ def prune_output_blocks_in_compound_match_query(compound_match_query):
         return CompoundMatchQuery(match_queries=match_queries)
 
 
+def _construct_location_to_predicate_list(match_query):
+    """Return a dict mapping location -> list of filter predicates applied at that location."""
```

Comment:
`filter predicates` is somewhat confusing and ill-defined. When you separate out the predicate expression from a `Filter` block, calling it a `predicate` is a bit questionable since it's not clear to what and where it applies.

I am also confused about why a list is necessary, when there is a lowering pass that merges filters that appear at the same location. Are we duplicating code here? That pass is very similar the code around line 624 here.